q2_moshpit/plugin_setup.py

```diff
@@ -404,6 +408,55 @@
                 'analyses.'
 )
 
+plugin.methods.register_function(
+    function=q2_moshpit.eggnog.build_custom_diamond_db,
+    inputs={
+        'sequences': FeatureData[ProteinSequence],
+        'taxonomy_data': ReferenceDB[TaxonomyNCBI],
+    },
+    input_descriptions={
+        'sequences': "Artifact containing protein reference database file "
+                     "in FASTA format.",
+        'taxonomy_data': "Artifact containing taxonomy data. "
+                         "Needed in order to provide taxonomy features. "
+                         "Can be generated through name_of_action."
+                         # TODO: update action name here
+    },
+    outputs=[('diamond_db', ReferenceDB[Diamond])],
+    output_descriptions={
+        'diamond_db': "Artifact containing a binary DIAMOND database file."
+    },
+    parameters={
+        "threads": Int % Range(1, None),
+        "verbose": Bool,
+        "log": Bool,
+        "file_buffer_size": Int % Range(1, None),
+        "ignore_warnings": Bool,
+        "no_parse_seqids": Bool
+    },
+    parameter_descriptions={
+        "threads": "Number of CPU threads. By default, the program will "
+                   "auto-detect and use all available virtual cores on the "
+                   "machine.",
+        "verbose": "Enable more verbose terminal output.",
+        "log": "Enable even more verbose terminal output, which is also "
```

Comment:
Sorry Michal, just to be clear, you would like me to remove the parameter and always set it to true? What should I do with the log file that I created? By default, it will create it in the current working directory. Should we leave it like that? I think it might be desirable to switch off this functionality no? Therefore it would be good to keep the parameter?

Reply:
Or should I include it as part of the output always and simply output an empty log when the parameter is set to false?